components/org.wso2.carbon.identity.mgt/src/main/java/org/wso2/carbon/identity/mgt/impl/util/UnitOfWork.java

```diff
@@ -87,6 +87,17 @@ public void endTransaction() throws SQLException {
         connection.commit();
     }
 
+    /**
+     * Revoke the transaction when catch then sql transaction errors.
+     *
+     * @param connection Database connection.
+     * @throws SQLException SQL Exception.
+     */
+    public static void rollBackTransaction(Connection connection) throws SQLException {
```

Comment:
We can handle the SQLException within this method and throw

throw new UniqueIdResolverException("Error occurred while transaction rollback. May be due to database or connection issue", e1);